driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/OperationExecutorImpl.java

```diff
@@ -95,22 +105,29 @@
             session.notifyOperationInitiated(operation);
         }
 
-        return clientSessionHelper.withClientSession(session, this)
-                .map(clientSession -> getReadWriteBinding(ReadPreference.primary(), readConcern, clientSession,
-                                                              session == null && clientSession != null))
-                .switchIfEmpty(Mono.fromCallable(() -> getReadWriteBinding(ReadPreference.primary(), readConcern, session, false)))
-                .flatMap(binding ->
-                        Mono.<T>create(sink -> operation.executeAsync(binding, (result, t) -> {
-                            try {
-                                binding.release();
-                            } finally {
-                                sinkToCallback(sink).onResult(result, t);
-                            }
-                        })).doOnError((t) -> {
-                            labelException(session, t);
-                            unpinServerAddressOnTransientTransactionError(session, t);
-                        })
-                );
+        return new Publisher<T>() {
+            @Override
+            public void subscribe(final Subscriber<? super T> s) {
+                // TODO: Extract a request context from the subscriber and add it to the binding
```

Comment:
this sounds like a typical use case for `Mono.deferContextual`, which would lazily create the `clientSessionHelper` mono with a knowledge of the `ContextView`.

Reply:
We are depending on Californium-SR23 currently, so don't have access to either `Mongo.deferContextual` or `ContextView`.  We could consider updating dependencies but at the time we took the dependency (not so long ago actually) this seemed safer.

Reply:
@mp911de which version of Reactor do other r2bc typically use?

I would warn against using `Californium` trains, which are now EoL. Jumping to `Dysprosium` would allow you to upgrade more easily while enabling use of `deferWithContext` (which deals with `Context` instead of `ContextView`), while jumping all the way up to `2020.0.x` (core 3.4.x) would put you on the latest and greatest, albeit with a little bit more migration work.

Reply:
R2DBC drivers use at least `Dysprosium`.

Reply:
When we update our minimum dependency we can consider this.

Reply:
FYI `Dysprosium` line (core 3.3.x) is going EoL by the end of the year too...

Reply:
Good to know.